### PR TITLE
platform: tigerlake: remove unused definitions

### PR DESCRIPTION
**Pull request by kv2019i**
platform: tigerlake: cleanup unused code for IPC3, SPI, GPIO, XTOS
platform: tigerlake: remove unused definitions

### CODE REVIEW

File: src/platform/intel/cavs/platform.c [hunk 1]
```diff
@@ -11,13 +11,10 @@
 #if (CONFIG_CAVS_LPS)
```
Comment (btian1) [inline src/platform/intel/cavs/platform.c:11]: will CONFIG_CAVS_LPS be set in one platform?

**Review by btian1 — APPROVED** [review]

File: src/platform/intel/cavs/platform.c [hunk 2]
```diff
@@ -322,14 +153,6 @@ int platform_init(struct sof *sof)
 	if (ret < 0)
 		return ret;
 
-	/* init low latency single channel DW-DMA domain and scheduler */
-	sof->platform_dma_domain =
-		dma_single_chan_domain_init
-			(&sof->dma_info->dma_array[PLATFORM_DW_DMA_INDEX],
-			 PLATFORM_NUM_DW_DMACS,
-			 PLATFORM_DEFAULT_CLOCK);
-	scheduler_init_ll(sof->platform_dma_domain);
```
Comment (lyakh) [inline src/platform/intel/cavs/platform.c:331]: this is removing functionality that isn't used by default and has a high, but not 100% chance of being broken. Still it's also possible at least in theory that someone is using it now, isn't it? So, it's probably a decision call - whether we want to deprecate it first and maybe disable by default or just remove it as you do it here

**Review by lyakh — APPROVED** [review]

File: src/platform/intel/cavs/platform.c (hunk 1 above)
Comment (kv2019i) [reply src/platform/intel/cavs/platform.c:11]: @btian1 I'll follow-up with separate PR to remove CAVS_LPS stuff. This needs a bit more surgery so I'll do that in a separate patch.

File: src/platform/intel/cavs/platform.c (hunk 2 above)
Comment (kv2019i) [reply src/platform/intel/cavs/platform.c:331]: @lyakh Good catch. I did go a bit back and forth on this. But in the end, cAVS2.5 IPC4 config is only made available as a opt-in release (with e.g. SOF2.6), so the usage at this point is very limited, so chances of somebody using it for something exotic like DMA scheduler domain, would seem much smaller than odds of winning in lottery. So I went ahead and removed.

**Comment by kv2019i** [comment]
Let me rebase and upload. One Intel internal test fails, but the fails are on platforms not touched in this PR, so seems like issues with the DUTs. Let me rebase.

**Pull request merged**